neuralmonkey/encoders/cnn_encoder.py

```diff
@@ -204,9 +204,10 @@ def feed_dict(self, dataset: Dataset, train: bool = False) -> FeedDict:
 
         f_dict = {}
         f_dict[self.image_input] = images / 225.0
-        # it is one everywhere where non-zero, i.e. zero columns are masked out
-        f_dict[self.image_mask] = np.sum(
-            np.sign(images), axis=3, keepdims=True)
+        # the image mask is one everywhere where the image is non-zero, i.e.
+        # zero pixels are masked out
+        f_dict[self.image_mask] = np.sign(
+            np.sum(images, axis=3, keepdims=True))
```

Comment:
Images je (batch, width, height, channels) a předpoklad tady je, že obsahem jsou buď čísla od 0 do 1 nebo samý nuly?

Jakej je rozdíl mezi černym pixelem a vymaskovanym pixelem? Je potřeba tam dělat rozdíl?

Taky mě teď začlo zajímat, proč to dělíš zrovna 225? Když už tak bych čekal 255.

Reply:
(tech 225 je na ř. 206)

Reply:
Každej černej pixel se vymaskuje, ale na žádný doopravický fotce fakt nulově černej pixel nikdy nebude a kdyby náhodou byl, tak bude někde jeden a ta maska se spraví na hned první konvoluční vrstvě, protože ta maska se smrskává maxpoolingem.

Ve skutečnosti myslím 255.